src/main/host/syscall_handler.c

```diff
@@ -229,12 +233,11 @@ static void _syscallhandler_post_syscall(SysCallHandler* sys, long number,
             scr = log_syscall(sys->process, thread_getID(sys->thread), #s, "...", scr);            \
         }                                                                                          \
         break
-
-#define HANDLE_RUST(s)                                                         \
-    case SYS_##s:                                                              \
-        _syscallhandler_pre_syscall(sys, args->number, #s);                    \
-        scr = rustsyscallhandler_##s(sys, args);                               \
-        _syscallhandler_post_syscall(sys, args->number, #s, &scr);             \
+#define HANDLE_RUST(s)                                                                             \
+    case SYS_##s:                                                                                  \
+        _syscallhandler_pre_syscall(sys, args->number, #s);                                        \
+        scr = rustsyscallhandler_syscall(sys->syscall_handler_rs, sys, args);                      \
+        _syscallhandler_post_syscall(sys, args->number, #s, &scr);                                 \
```

Comment:
This should change to the following so that we don't accidentally use `syscall_handler_rs` twice:

```c
{
    _syscallhandler_pre_syscall(sys, args->number, #s);
    SyscallHandler* handler = sys->syscall_handler_rs;
    sys->syscall_handler_rs = NULL;
    scr = rustsyscallhandler_syscall(handler, sys, args);
    sys->syscall_handler_rs = handler;
    _syscallhandler_post_syscall(sys, args->number, #s, &scr); 
}
```

(and below for `SYS_fcntl64`)

Reply:
Done!